cpp/src/join/join.cu

```diff
@@ -51,43 +83,69 @@ std::unique_ptr<table> inner_join(
+    auto join_indices = hj_obj.inner_join(right.select(right_on), compare_nulls, stream, mr);
+    std::unique_ptr<table> left_result  = detail::gather(left,
+                                                        join_indices.second->begin(),
+                                                        join_indices.second->end(),
+                                                        out_of_bounds_policy::DONT_CHECK,
+                                                        stream,
+                                                        mr);
+    std::unique_ptr<table> right_result = detail::gather(right,
+                                                         join_indices.first->begin(),
+                                                         join_indices.first->end(),
+                                                         out_of_bounds_policy::DONT_CHECK,
+                                                         stream,
+                                                         mr);
+    return combine_table_pair(std::move(left_result), std::move(right_result));
   } else {
-    cudf::hash_join hj_obj(right, right_on, compare_nulls, stream);
-    auto probe_build_pair = hj_obj.inner_join(left,
-                                              left_on,
-                                              columns_in_common,
-                                              cudf::hash_join::common_columns_output_side::PROBE,
-                                              compare_nulls,
-                                              stream,
-                                              mr);
-    return cudf::detail::combine_table_pair(std::move(probe_build_pair.first),
-                                            std::move(probe_build_pair.second));
+    cudf::hash_join hj_obj(right.select(right_on), compare_nulls, stream);
+    auto join_indices = hj_obj.inner_join(left.select(left_on), compare_nulls, stream, mr);
+    std::unique_ptr<table> left_result  = detail::gather(left,
+                                                        join_indices.first->begin(),
+                                                        join_indices.first->end(),
+                                                        out_of_bounds_policy::DONT_CHECK,
+                                                        stream,
+                                                        mr);
+    std::unique_ptr<table> right_result = detail::gather(right,
+                                                         join_indices.second->begin(),
+                                                         join_indices.second->end(),
+                                                         out_of_bounds_policy::DONT_CHECK,
+                                                         stream,
+                                                         mr);
+    return combine_table_pair(std::move(left_result), std::move(right_result));
```

Comment:
Instead of repeating logic that's in the `cudf::inner_join` that returns gather maps, couldn't you just call that other overload here?

Reply:
Fixed - I think

Reply:
ditto below